pytensor/link/jax/dispatch/sparse.py

```diff
@@ -0,0 +1,38 @@
+import jax.experimental.sparse as jsp
+from scipy.sparse import spmatrix
+
+from pytensor.link.jax.dispatch import jax_funcify, jax_typify
+from pytensor.sparse.basic import Dot, SparseVariable, StructuredDot
+from pytensor.sparse.sharedvar import SparseTensorSharedVariable
+from pytensor.sparse.type import SparseTensorType
+
+
+@jax_typify.register(spmatrix)
+def jax_typify_spmatrix(matrix, dtype=None, **kwargs):
+    # Note: This changes the type of the constants from CSR/CSC to BCOO
+    # We could add BCOO as a PyTensor type but this would only be useful for JAX graphs
+    # and it would break the premise of one graph -> multiple backends.
+    # The same situation happens with RandomGenerators...
+    return jsp.BCOO.from_scipy_sparse(matrix)
+
+
+@jax_funcify.register(Dot)
+@jax_funcify.register(StructuredDot)
+def jax_funcify_sparse_dot(op, node, **kwargs):
+    for input in node.inputs:
+        if isinstance(input, (SparseVariable, SparseTensorSharedVariable)):
+            raise NotImplementedError(
+                "JAX sparse dot only implemented for constant sparse inputs"
+            )
```

Comment:
you mean constant sparse inputs, as opposed a sparse matrix where the non-zero elements are TensorVariables?

Reply:
That is what I understood (so it would work for splines, HSGP by groups, and many other things that can be expressed this way)

Reply:
Oh my comment was just that instead of checking if `isinstance(input, (Variable, SharedVariable)) ` I could check `not isinstance(input, Constant)` which is a bit more clear